doc/bundle_lookup_matcher.md

```diff
@@ -0,0 +1,132 @@
+);
+
+var enUs = new Cldr( "en-US" );
+console.log( enUs.attributes.bundle ); // "en-US"
+console.log( enUs.main( "dates/calendars/gregorian/dateFormats/short" ) ); // "M/d/yy"
+
+var enGb = new Cldr( "en-GB" );
+console.log( enGb.attributes.bundle ); // "en-GB"
+console.log( enGb.main( "dates/calendars/gregorian/dateFormats/short" ) ); // "dd/MM/y"
+```
+
+When instances are created, its `.attributes.bundle` reveals the matched bundle. The `.main` method uses this information to traverse the correct main item.
+
+What happens if we include `main/en/ca-gregorian` to the above example? 
+
+```javascript
+Cldr.load(
+  cldr( "supplemental/likelySubtags" ),   // JSON data from supplemental/likelySubtags.json
+  cldr( "main/en/ca-gregorian" ),         // JSON data from main/en/ca-gregorian.json
+  cldr( "main/en-US/ca-gregorian" ),      // JSON data from main/en-US/ca-gregorian.json
+  cldr( "main/en-GB/ca-gregorian" )       // JSON data from main/en-GB/ca-gregorian.json
+);
+
+var enUs = new Cldr( "en-US" );
+console.log( enUs.attributes.bundle ); // "en"
+console.log( enUs.main( "dates/calendars/gregorian/dateFormats/short" ) ); // "M/d/yy"
+
+var enGb = new Cldr( "en-GB" );
+console.log( enGb.attributes.bundle ); // "en-GB"
+console.log( enGb.main( "dates/calendars/gregorian/dateFormats/short" ) ); // "dd/MM/y"
+```
+
+Now, the `en-US` requested locale uses the `en` bundle (not the `en-US` bundle as used in the first example) and `en-GB` still uses the `en-GB` bundle. Why? Because, `en` is the default content for `en-US` (deduced from likelySubtags data). Default content means that the child content is all in the parent. Therefore, both `en` and `en-US` are identical. Our bundle lookup matching algorithm always picks the grandest available parent. Note the retrieved main item is still the correct one (as it should be).
+
+A good observer may notice that loading both `main/en/ca-gregorian` and `main/en-US/ca-gregorian` is redundant. Although loading both is not a problem, loading either the `en` or the `en-US` bundle alone is enough.
+
+Let's add a bit of sugar to the requested locales.
+
+```javascript
+var en = new Cldr( "en" );
+console.log( en.attributes.bundle ); // "en"
+
+var enUs = new Cldr( "en-US" );
+console.log( enUs.attributes.bundle ); // "en"
+
+var enLatnUs = new Cldr( "en-Latn-US" );
+console.log( enLatnUs.attributes.bundle ); // "en"
+```
+
+All instances above obviously matches the same `en` bundle. Because, (a) `en` is the default content for `en-US` and (b) `en-US` is the default content for `en-Latn-US`.
+
+What happens if the requested locale includes unicode extensions?
+
+```javascript
+var en = new Cldr( "en-US-u-cu-USD" );
+console.log( en.attributes.bundle ); // "en"
+console.log( en.main( "numbers/currencies/{u-cu}/displayName" ) ); // "US Dollar"
+```
+
+Unicode extensions are obviously ignored on bundle lookup. Note they are accessible via variable replacements.
+
+Follow other non-obvious lookups.
```

Comment:
`zh-Hant` is the default script for `zh-TW`. Those are just non-obvious lookups. Because, it's not a direct map "requested locale" to "bundle locale".